moped-editor/src/views/projects/projectView/ProjectPhaseForm.js

```diff
@@ -0,0 +1,292 @@
+import { useEffect } from "react";
+import { useMutation } from "@apollo/client";
+import { useForm } from "react-hook-form";
+import Alert from "@mui/material/Alert";
+import Button from "@mui/material/Button";
+import CheckCircle from "@mui/icons-material/CheckCircle";
+import CircularProgress from "@mui/material/CircularProgress";
+import FormControl from "@mui/material/FormControl";
+import FormHelperText from "@mui/material/FormHelperText";
+import Grid from "@mui/material/Grid";
+import { yupResolver } from "@hookform/resolvers/yup";
+import ControlledAutocomplete from "src/components/forms/ControlledAutocomplete";
+import ControlledDateField from "src/components/forms/ControlledDateField";
+import ControlledTextInput from "src/components/forms/ControlledTextInput";
+import ControlledCheckbox from "src/components/forms/ControlledCheckbox";
+import ControlledSwitch from "src/components/forms/ControlledSwitch";
+import {
+  phaseValidationSchema,
+  onSubmitPhase,
+  useDefaultValues,
+  useSubphases,
+  useCurrentPhaseIdsToClear,
+} from "./ProjectPhase/helpers";
+import { useResetDependentFieldOnParentFieldChange } from "./ProjectComponents/utils/form";
+import { UPDATE_PROJECT_PHASE, ADD_PROJECT_PHASE } from "src/queries/project";
+
+const ProjectPhaseForm = ({
+  phase,
+  phases,
+  currentProjectPhaseIds,
+  onSubmitCallback,
+}) => {
+  const isNewPhase = !phase.project_phase_id;
+
+  const defaultValues = useDefaultValues(phase);
+
+  /** initiatlize react hook form with validation */
+  const {
+    handleSubmit,
+    control,
+    watch,
+    setValue,
+    formState: { isDirty, errors: formErrors },
+  } = useForm({
+    defaultValues,
+    resolver: yupResolver(phaseValidationSchema),
+  });
+
+  const subphases = useSubphases(watch("phase_id"), phases);
+
+  const isCurrentPhase = watch("is_current_phase");
+
+  useResetDependentFieldOnParentFieldChange({
+    parentValue: watch("phase_id"),
+    dependentFieldName: "subphase_id",
+    valueToSet: null,
+    setValue,
+  });
+
+  const [mutate, mutationState] = useMutation(
+    isNewPhase ? ADD_PROJECT_PHASE : UPDATE_PROJECT_PHASE
+  );
+
+  const currentPhaseIdsToClear = useCurrentPhaseIdsToClear(
+    phase.project_phase_id,
+    isCurrentPhase,
+    currentProjectPhaseIds
+  );
+
+  const [phase_start, phase_end] = watch(["phase_start", "phase_end"]);
+
+  /**
+   * Defaults is_phase_start_confirmed to true if date is today or before
```

Comment:
One question that I thought about when testing the UI, do we need to enforce the checking of the box when the start/end date is today or before? To put it another way, should users be able to uncheck the confirmed box after this hook checks it?

Reply:
+1  Chia: that was my thinking, too